File: dsa_leetcode_beginner/0191-number-of-1-bits/0191-number-of-1-bits.java

public class NumberOf1Bits {
    public static int hammingWeight(int n) {
        int count = 0;

        // Assuming a 32-bit integer
        for (int i = 0; i < 32; i++) {
            count += (n & 1);
            n = n >> 1;
        }

        return count;
    }

    public static void main(String[] args) {
        int n = 11;  // Example input (binary: 1011)
        System.out.println(hammingWeight(n));  // Output: 3
    }
}

File: dsa_leetcode_beginner/0191-number-of-1-bits/0191-number-of-1-bits.py
class Solution:
    def hammingWeight(self, n: int) -> int:
        count = 0

        # Assuming a 32-bit integer
        for _ in range(32):  
            # Check the LSB
            count = count + (n & 1) 

            # Right shift the bits of n
            n = n >> 1                

        return count
    
    
#Question: https://leetcode.com/problems/number-of-1-bits
#Blog: https://blog.unwiredlearning.com/number-of-1-bits